python-threatexchange/threatexchange/cli/cli_config.py

```diff
@@ -0,0 +1,51 @@
+#!/usr/bin/env python
+# Copyright (c) Facebook, Inc. and its affiliates. All Rights Reserved
+
+import typing as t
+import pathlib
+import os
+
+from threatexchange.collab_config import CollaborationConfig
+
+"""
+Local storage and configuration for the CLI.
+
+The CLI and Hasher-Matcher-Actioner are roughly parallel, but this isn't a 
+scalable service running on AWS. Instead, we have all of our state in
+a file (likely ~/.threatexchange)
+"""
+
+FETCH_STATE_DIR_NAME = "fetched_state"
+COLLABORATION_CONFIG_DIR_NAME = "collaborations"
+INDEX_STATE_DIR_NAME = "index"
+CONFIG_FILENAME = "config.json"
+
+
+class CliState:
+    """
+    A wrapper around stateful information stored for the CLI.
+
+    Everything is just in a single file (usually ~/.threatexchange).
+    """
+
+    def __init__(self, dir: pathlib.Path):
+        assert dir.is_file
+        self._dir = dir
+
+    def get_collab_by_name(self) -> t.Optional[CollaborationConfig]:
+        """
+        Gets only a single stored collaboration config by its given name.
+        """
+        return None
+
+    def update_collab(self, collab: CollaborationConfig, delete: bool = False) -> None:
```

Comment:
nit: not a lot more effort to create a dedicated `delete_collab` method. I find the usage of `delete` as an argument confusing.

Reply:
Also, question: Can collab names be changed?

If they can't, then all is well.

If collab.name can be updated, though, how would the implementation figure out the collab to be updated?

Reply:
Yea, plan is for names to be immutable, because I actually need to key them several places in the system (index). The alternative would be generating IDs for collabs, but then I have ID assignment problems, so seemed easier to just make the collab name immutable, but also was going to see what happened when I did some of the later diffs.